src/main.rs

```diff
@@ -192,19 +193,42 @@ async fn on_connect(socket: SocketRef, Data(auth): Data<AuthPayload>, pool: Stat
     socket.on(
         "attack",
         |socket: SocketRef, Data::<[usize; 2]>([i, j]), pool: State<PgPool>| async move {
-            let (hit, sunk) = match attack(socket.id, (i, j), &pool).await {
+            let (hit, sunk, winner_loser) = match attack(socket.id, (i, j), &pool).await {
                 Ok(res) => res,
                 Err(e) => {
                     tracing::error!("{:?}", e);
                     return;
                 }
             };
-            tracing::info!("Attacking at: ({}, {}), result: {:?}", i, j, hit);
+            let binding = "".to_string();
+            let winner_loser = winner_loser
+                .as_ref()
+                .map(|(w, l)| (w, l))
+                .unwrap_or((&binding, &binding));
+
+            tracing::info!(
+                "Attacking at: ({}, {}), result: {:?}, winner: {:?}, loser: {:?}",
+                i,
+                j,
+                hit,
+                winner_loser.0,
+                winner_loser.1
+            );
+
             socket
                 .within(socket.rooms().unwrap().first().unwrap().clone())
                 .emit(
                     "attacked",
-                    serde_json::json!({"by": socket.id.as_str(), "at": [i, j], "hit": hit, "sunk": sunk}),
+                    serde_json::json!(
+                        {
+                            "by": socket.id.as_str(),
+                            "at": [i, j],
+                            "hit": hit,
+                            "sunk": sunk,
+                            "winner": winner_loser.0,
+                            "loser": winner_loser.1
```

Comment:
We can just emit `has_won` which can be a bool. Frontend can determine if its them or not using `by` value.

Reply:
cool